pkg/skaffold/test/custom/custom.go

```diff
@@ -147,3 +153,64 @@ func (ct *Runner) TestDependencies() ([]string, error) {
 	}
 	return nil, nil
 }
+
+func (ct *Runner) retrieveCmd(ctx context.Context, out io.Writer, command string, bRes []build.Artifact) (*exec.Cmd, error) {
+	var cmd *exec.Cmd
+	// We evaluate the command with a shell so that it can contain env variables.
+	if runtime.GOOS == Windows {
+		cmd = exec.CommandContext(ctx, "cmd.exe", "/C", command)
+	} else {
+		cmd = exec.CommandContext(ctx, "sh", "-c", command)
+	}
+	cmd.Stdout = out
+	cmd.Stderr = out
+
+	env, err := ct.setEnv(bRes)
+	if err != nil {
+		return nil, fmt.Errorf("setting env variables: %w", err)
+	}
+	cmd.Env = env
+
+	dir, err := testContext(ct.testWorkingDir)
+	if err != nil {
+		return nil, fmt.Errorf("getting context for test: %w", err)
+	}
+	cmd.Dir = dir
+
+	return cmd, nil
+}
+
+func (ct *Runner) setEnv(bRes []build.Artifact) ([]string, error) {
```

Comment:
probably `getEnv` is more apt than `setEnv` here.

Reply:
Renamed the method to getEnv().